site-src/geps/gep-1364.md

```diff
@@ -0,0 +1,328 @@
+are intended more for human consumption than machine consumption.
+
+This means that there's no guidance from upstream about condition polarity. We'll
+discuss this more when we talk about new conditions.
+
+The guidance about Conditions being added as soon as a controller sees a resource
+is a bit unclear - as written in the conventions, it seems to imply that _all_ 
+relevant conditions should always be added, even if their status has to be set to
+`unknown`.
+Gateway API resources do not currently require this, and the practice seems to be
+uncommon.
+
+## Proposed changes
+
+### Proposed changes summary
+
+* All the current Conditions that indicate that the resource is okay and ready
+for processing witll be replaced with `Attached`, except for GatewayClass (since
+it is the root of the resource tree, it will stay with `Accepted`).
+* In general, resources should be considered `Attached` if their config will
+generate some config in the underlying data plane.
+* All Conditions will be positive polarity (`GoodState: true`, not `BadState: false`)
+* All relevant Conditions for a resource must be added when it's observed.
+For example, HTTPRoutes must always have `Accepted`, `Ready`, and `ResolvedRefs`
+conditions, regardless of their state.
+
+The exact list of changes is detailed below. The next few sections detail
+the reasons for these large-scale changes.
+
+### Conceptual and language changes
+
+Gateway API resources are, conceptually, all about breaking up the configuration for a
+data plane into separate resources that are _expressive_ and _extensible_, while being
+split up along _role-oriented_ boundaries.
+
+So, when we talk about Gateway API, it's _always_ about a _system of related resources_.
+
+We already acknowledge this when we talk about Routes "attaching" to Gateways, or Gateways
+referencing Services, or Gateways requiring a GatewayClass in their spec.
+
+So, this GEP proposes that we formalize this by leaning into the word "attach" as the
+primary verb to describe the relationships between resources in the system.
+
+So, Gateways _attach_ to GatewayClasses, Routes _attach_ to Gateways, and in a
+bit of an extension, ReferenceGrants _attach_ to the things they are Granting
+Reference access to.
+
+Of course, because we're using all of this configuration to describe some sort of data
+path from "outside"/lacking cluster context to "inside"/enriched with cluster context,
+we also need a way to describe when that data path is configured and working.
+
+Again, we already have a word we use for this - "Ready", so this GEP proposes keeping
+that and tightening up the definition.
+
+### Condition polarity
+
+In terms of the polarity of conditions, we have three options, of which only two are
+really viable:
+* All conditions must be negative polarity
+* All conditions must be positive polarity
+* Some conditions can be positive polarity, but most should be negative.
```

Comment:
My own personal preference would be for conditions with a positive polarity to represent the normal healthy state and always be present, and for conditions with a negative polarity to represent very specific error states and to only be present when true. I think we should do everything we can to avoid the presence of double negatives like "Detached == False", but recognize that there are some error states that are best represented with a negative condition. I just don't think those negative conditions should be present when false.

Reply:
If we're going to have a mixed-polarity set of conditions, then I think we need to define:
- which Conditions are positive polarity (Probably `Attached`, `Programmed`, `Ready`, and `Valid` if we do it)
- which Conditions are positive polarity and should always be present (Probably `Attached`, `Programmed`, and `Valid` if we do it).
- which Conditions are core (currently, everything other than `Ready`)
- which Conditions are extended (only `Ready`).
- which Conditions are negative (all the error conditions).